src/nyc_trees/apps/home/training/types.py

```diff
@@ -199,25 +199,66 @@ def extract_answers(cls, post_fields):
     def score(self, answers):
         """
         How many correct answers are selected?
-        answers - dict of {question_index => answer_index, ...}
+        answers - dict of {question_index => answer_indexes, ...}
         """
         result = 0
         for i, question in enumerate(self.questions):
-            if answers.get(i, -1) == question.answer:
+            answer = answers.get(i, None)
+            if question.is_correct(answer):
                 result += 1
         return result
 
 
-class Question(object):
+class AbstractQuestion(object):
```

Comment:
would it make more sense to represent a single choice question as a special case of a multiple choice question in which the set of correct answers has one member? Then you can do similar set operations for both types and probably just use one class.